coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -33,3 +41,107 @@ def __init__(self, documentation, language, docstyle,
 
     def __str__(self):
         return self.documentation
+
+    def parse_documentation(self):
+        """
+        Parses documentation independent of language and docstyle.
+
+        :return:
+            A list of the parsed documentation metadata.
+        :raises:
+            NotImplementedError
+        """
+        if self.language == "python":
+            return self.parse_python()
+        else:
+            raise NotImplementedError(
+                "Documentation parsing for {} in {}"
+                " has not been implemented yet".format(
+                    self.language, self.docstyle))
+
+    def parse_python(self):
+        """
+        Parse documentation. Usable attributes are:
+
+        ``:param``
+        ``@param``
+
+        ``:return``
+        ``@return``
+
+        :return: A list of parsed sections(descriptions, params, return values)
+        """
+        lines = self.documentation.splitlines(keepends=True)
+
+        param_identifiers = (":param ", "@param ")
+        return_identifiers = (":return: ", "@return: ")
+
+        return self._parse_documentation(lines, param_identifiers,
+                                         return_identifiers)
+
+    def _parse_documentation(self, lines, params, returns):
```

Comment:
maybe another name?

> Parses documentation based on parameter and return symbols.

So you could maybe a bit more specific, like `_parse_documentation_with_symbols` or something like that